### PR TITLE
Add tests for liveness of NodeLists/HTMLCollections

### LINKED ISSUE #1505: Add tests for liveness of NodeLists/HTMLCollections.
(Please add any I missed.)
- [ ] Node#childNodes
- [ ] ParentNode#children
- [ ] {Document,Element}#getElementsByTagName
- [ ] {Document,Element}#getElementsByTagNameNS
- [ ] {Document,Element}#getElementsByClassName
- [ ] Document#images
- [ ] Document#embeds
- [ ] Document#plugins
- [ ] Document#links
- [ ] Document#forms
- [ ] Document#scripts
- [ ] Document#commands
- [ ] Document#getElementsByName
- [ ] Document#getItems

### PR DESCRIPTION
**Pull request by xiuqijix**
Add following tests (Fix #1505):

 - Node#childNodes
 - ParentNode#children
 - {Document,Element}#getElementsByTagName
 - {Document,Element}#getElementsByTagNameNS
 - {Document,Element}#getElementsByClassName
 - Document#images
 - Document#embeds
 - Document#plugins
 - Document#links
 - Document#forms
 - Document#scripts
 - Document#getElementsByName

@Ms2ger, PTAL.

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/22635)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/327579392)

**Comment by foolip** [comment]
From https://pulls.web-platform-tests.org/build/22635 it looks like all browsers pass all of these tests. That's a bit surprising, are things here really perfectly interoperable? @Ms2ger, when you filed https://github.com/w3c/web-platform-tests/issues/1505, were you aware of anything in particular in need of testing?

**Comment by gsnedders** [comment]
@foolip for most of these, much of the web would be badly broken if they weren't live, so I'm not surprised by interop.

**Comment by Ms2ger** [comment]
I don't recall if there were any bugs I knew of at that time; I think I would have said so if I did.

**Comment by kaixinjxq** [comment]
@Ms2ger, any comments on my tests?

**Comment by Ms2ger** [comment]
Haven't had time to review them yet.

**Comment by plehegar** [comment]
I wonder if it's worth testing childNodes with an innerHTML manipulation.

**Comment by foolip** [comment]
@tkent-google, is this something you could review?

**Review by tkent-google — COMMENTED** [review]
I prefer adding liveness tests to each of API test files to adding a single liveness test file for various API.

e.g. The first test() in Node-collection-live.html should be in dom/nodes/Node-childNodes.html.

**Comment by kaixinjxq** [comment]
@tkent-google, sorry for late response, I have addressed your comments in new commit, PTAL.

File: dom/nodes/Element-getElementsByClassName.html [hunk 1]
```diff
@@ -18,4 +18,23 @@
   assert_false(list instanceof NodeList, "NodeList")
   assert_true(list instanceof HTMLCollection, "HTMLCollection")
 }, "Interface should be correct.")
+test(function() {
```
Comment (tkent-google) [inline dom/nodes/Element-getElementsByClassName.html]: Add a blank line before this line.

File: dom/nodes/Document-Element-getElementsByTagName.js [hunk 2]
```diff
@@ -190,4 +190,19 @@ function test_getElementsByTagName(context, element) {
...
+    var l = context.getElementsByTagName("abc");
+    assert_true(l instanceof HTMLCollection);
+    assert_equals(l.length, 1);
+
+    var t2 = element.appendChild(document.createElement("abc"));
+    assert_equals(l.length, 2);
+
+    element.removeChild(t2);
+    assert_equals(l.length, 1);
+  }, "getElementsByTagName() return an HTMLCollection is liveness");
```
Comment (tkent-google) [inline dom/nodes/Document-Element-getElementsByTagName.js]: The test description looks weird grammatically.
Should it be "getElementsByTagName() should be a live collection" or something?

This comment should be applied to all of new tests.

File: dom/nodes/Element-getElementsByClassName.html (hunk 1 above)
Comment (kaixinjxq) [reply dom/nodes/Element-getElementsByClassName.html]: Updated

File: dom/nodes/Document-Element-getElementsByTagName.js (hunk 2 above)
Comment (kaixinjxq) [reply dom/nodes/Document-Element-getElementsByTagName.js]: Thanks, @tkent-google, updated the test descriptions.

**Review by tkent-google — APPROVED** [review]

**Pull request merged**